v63003/u_inref/gtm8844.csh

```diff
@@ -0,0 +1,116 @@
++^Y -name=triggery -command=set -xecute="do triggerfny^gtm8844"
+EOF
+
+echo "# Running functions that violate our restrictions"
+echo "# Expect two GTM_FATAL files at the bottom of the reference file"
+echo "# -------------------------------------------------------------------------------------------"
+echo "# Two consecutive halts"
+$ydb_dist/mumps -run haltfn^gtm8844
+ls *FATAL*
+rm *FATAL*
+echo "# -------------------------------------------------------------------------------------------"
+echo "# Two consecutive zhalts"
+$ydb_dist/mumps -run zhaltfn^gtm8844
+ls *FATAL*
+rm *FATAL*
+echo "# -------------------------------------------------------------------------------------------"
+echo "# Confirming ZHALT,ZGOTO 0 produces an error in the shell"
+$ydb_dist/mumps -run zgotofn^gtm8844
+echo "Status After ZGOTOFN=$status"
+if ($status) then
+	echo "Error Detected"
+endif
+echo "# -------------------------------------------------------------------------------------------"
+$MULTISITE_REPLIC_PREPARE 2
+setenv gtm_test_repl_norepl 1
+$gtm_tst/com/dbcreate.csh mumps 9 >>& dbcreate.out
+$MSR START INST1 INST2 >>& MSRStart.out
+
+# Creating initial values for global variables X and Y, uploading triggers
+$MSR RUN INST1 '$ydb_dist/mumps -run ^%XCMD "set ^X=0  set ^Y=0"'
+$MUPIP trigger -triggerfile=trigger.txt
+
+echo ""
+
+echo "# Test for ZHALT in a trigger"
+# We use the global variable ^H in HREG (unreplicated in instance 2) to ensure that when we turn the database back on, the update in the backlog
+# can go through without error
+$MSR RUN INST2 '$ydb_dist/mumps -run ^%XCMD "set ^H=0;write ^H"' >>& extrastuff.out
+echo "# Setting off Trigger"
+$MSR RUN INST1 '$ydb_dist/mumps -run ^%XCMD "set ^X=1"'
+echo "# Check primary"
+$MSR RUN INST1 '$ydb_dist/mumps -run ^%XCMD "write ^X"'
+echo "# Check secondary"
+$MSR RUN INST2 '$ydb_dist/mumps -run ^%XCMD "write ^X,!"'
+$MSR RUN INST2 '$MUPIP replic -receiver -checkhealth >& checkhealth.tmp ; cat checkhealth.tmp' >& checkhealth.out
+set updprocpid = `$tst_awk '/PID.*Update process/{print $2}' checkhealth.out`
+$MSR RUN INST2 'set msr_dont_trace ; $gtm_tst/com/wait_for_proc_to_die.csh' $updprocpid
+$MSR RUN INST2 '$MUPIP replic -receiver -checkhealth >& checkhealth.tmp ; cat checkhealth.tmp' >& checkhealth.out
+cat checkhealth.out
+echo "# Restore the update process"
+# Using outx because we are expecting errors
+$MSR STOPRCV INST1 INST2 >>& restart.outx
+$MSR RUN INST2 '$ydb_dist/mumps -run ^%XCMD "set ^H=1;write ^H"'>>& restart.outx
+$MSR STARTRCV INST1 INST2 >>& restart.outx
+$MSR SYNC INST1 INST2 >>& restart.outx
+echo "# Confirm secondary's update process is restored and database is up to date"
+$MSR RUN INST2 '$MUPIP replic -receiver -checkhealth'
+$MSR RUN INST2 '$ydb_dist/mumps -run ^%XCMD "write ^X,!"'
+
+echo "# -------------------------------------------------------------------------------------------"
+
+echo "# Test for HALT in a trigger"
```

Comment:
The HALT and ZHALT logic seems to be almost a duplicate of each other. Can we merge this into a loop.